src/main/java/edu/harvard/iq/dataverse/ingest/IngestServiceBean.java

```diff
@@ -503,6 +503,89 @@ public int compare(DataFile d1, DataFile d2) {
         }
     }
     
+    public String startIngestJobForSingleFile(DataFile dataFile, AuthenticatedUser user) {
```

Comment:
Now that we have this function, should we simplify `startIngestJobs` to be

    public void startIngestJobs(Dataset dataset, AuthenticatedUser user) {
        for (DataFile dataFile : dataset.getFiles()) {
            if (dataFile.isIngestScheduled()) {
               startIngestJobForSingleFile(dataFile, user);
            }
        }
    }

Reply:
Actually, no - that would result in a different behavior. When a user uploads 10 ingestable files, startIngestJobs will schedule the ingest as one queued JMS job, for the entire batch of 10 files. We want to keep the dataset locked for as long as the entire batch is being processed. With 10 individual queued jobs this would be harder to control. The UNF recalculation would be slightly trickier too; there may be other potential issues. 

At the same time, these 2 methods do have more duplicated code than strictly necessary. I could definitely get away with using the old one (startIngestJobs) for  both ingest and reingest, with just slight modifications. I just felt like writing a method from scratch, for some reason.

As a reviewer, if you feel this is ugly/potentially confusing to a future developer, I'm willing to quickly do that and merge these 2 methods into one.

Reply:
If they were both nice small functions, I'd say don't bother, but this adds another ~100 lines of duplicate complicated code, so I think it's probably a good idea.

Reply:
OK, done. There's now a startIngestJobs(); and an extra small startIngestJobsForDataset(), that doesn't duplicate any code.